File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        stack<char>st;
        int l = s.size();
        for(int i=0;i<l;i++){
            if(s[i] == 'B'){
                if(!st.empty() && st.top()=='A'){
                    cout << "h" << endl;
                    st.pop();
                }
                else{
                    st.push(s[i]);
                }
            }
            else if(s[i] == 'D'){
                if(!st.empty() && st.top()=='C'){
                    cout << "th" << endl;
                    st.pop();
                }
                else{
                    st.push(s[i]);
                }
            }
           else{
               cout << s[i] << endl;
               st.push(s[i]);
           }
        }
        return st.size();
    }
};

File: 2696-minimum-string-length-after-removing-substrings/2696-minimum-string-length-after-removing-substrings.java
class Solution {
    public int minLength(String s) {
        Stack<Character> st = new Stack<>();
        
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == 'B'){
                if(st.size() > 0 && st.peek() == 'A'){
                    st.pop();
                }
                else{
                    st.add(s.charAt(i));
                }
            }
            else if(s.charAt(i) == 'D'){
                if(st.size() > 0 && st.peek() == 'C'){
                    st.pop();
                }
                else{
                    st.add(s.charAt(i));
                }
            }
            
            else{
                st.add(s.charAt(i));
            }
        }
        
        return st.size();
    }
}